amgut/connections.py

```diff
@@ -0,0 +1,50 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The American Gut Project Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+#
+# This module handles lazy-loading the various connections needed. This
+# simplifies code that relies on amgut (like scripts/ag) but that does not need
+# connections to the database since it handles them itself.
+
+
+class _LazyAgData(object):
```

Comment:
This is a very clean way to handle these connections. @josenavas @squirrelo, just pinging the two of you as it might be nice to adopt something like this in qiita